WeakAuras/BossMods.lua

```diff
@@ -0,0 +1,1508 @@
+      trigger.use_message and trigger.message or "",
+      trigger.use_message and trigger.message_operator or "",
+      trigger.use_cloneId and "true" or "false",
+      trigger.use_extend and tonumber(trigger.extend or 0) or 0,
+      trigger.use_remaining and "true" or "false",
+      trigger.remaining and tonumber(trigger.remaining or 0) or 0,
+      trigger.remaining_operator or "<"
+    )
+  end,
+  statesParameter = "full",
+  args = {
+    {
+      name = "spellId",
+      display = L["Key"],
+      type = "string",
+      store = true,
+      conditionType = "string"
+    },
+    {
+      name = "message",
+      display = L["Message"],
+      type = "longstring",
+      store = true,
+      conditionType = "string"
+    },
+    {
+      name = "remaining",
+      display = L["Remaining Time"],
+      type = "number",
+    },
+    {
+      name = "extend",
+      display = L["Offset Timer"],
+      type = "string",
+    },
+    {
+      name = "count",
+      display = L["Count"],
+      desc = L["Occurrence of the event, reset when aura is unloaded\nCan be a range of values\nCan have multiple values separated by a comma or a space\n\nExamples:\n2nd 5th and 6th events: 2, 5, 6\n2nd to 6th: 2-6\nevery 2 events: /2\nevery 3 events starting from 2nd: 2/3\nevery 3 events starting from 2nd and ending at 11th: 2-11/3\n\nOnly if DBM shows it on it's bar"],
+      type = "string",
+      conditionType = "string",
+    },
+    {
+      name = "cloneId",
+      display = L["Clone per Event"],
+      type = "toggle"
+    },
+    {
+      name = "note",
+      type = "description",
+      display = "",
+      text = function()
+        local ret = L["Note: This trigger will use BigWigs or DBM, in that order if both are installed.\n\nActive boss mod addon: "]
+        if Private.ExecEnv.BossMods.BigWigs.isInstalled then
+          ret = ret .. "BigWigs"
+        elseif Private.ExecEnv.BossMods.DBM.isInstalled then
+          ret = ret .. "DBM"
+        else
+          ret = ret .. L["none detected"]
+        end
+        return ret
+      end
+    },
+  },
+  automaticrequired = true,
+}
+Private.category_event_prototype.addons["Boss Mod Timer"] = L["Boss Mod Timer"]
+
+-- deprecated functions for compatibility for custom coded auras
+WeakAuras.GetAllDBMTimers = function() return Private.ExecEnv.BossMods.DBM:GetAllTimers() end
+WeakAuras.GetDBMTimerById = function(...) return Private.ExecEnv.BossMods.DBM:GetTimerById(...) end
+WeakAuras.GetDBMTimer = function(...) return Private.ExecEnv.BossMods.DBM:GetTimer(...) end
+WeakAuras.GetBigWigsTimerById = function(...) return Private.ExecEnv.BossMods.BigWigs:GetTimerById(...) end
+WeakAuras.GetAllBigWigsTimers = function() return Private.ExecEnv.BossMods.BigWigs:GetAllTimers() end
+WeakAuras.GetBigWigsStage = function(...) return Private.ExecEnv.BossMods.BigWigs:GetStage(...) end
+WeakAuras.RegisterBigWigsTimer = function() end
+WeakAuras.RegisterDBMCallback = function() end
```

Comment:
While the no-op funcs prevent errors due to calling nonexistent functions, custom coded aura calling them only works if the events it expects have been registered by other triggers. Ignore if that's intentional.